File: plugin/src/main/java/io/xol/dogez/plugin/player/PlayerProfiles.java
package io.xol.dogez.plugin.player;

import java.util.ArrayList;
import java.util.List;

import io.xol.chunkstories.api.player.Player;
import io.xol.dogez.plugin.XolioZGamemodePlugin;

public class PlayerProfiles {

	private final XolioZGamemodePlugin plugin;

	public PlayerProfiles(XolioZGamemodePlugin plugin) {
		this.plugin = plugin;
	}

	List<PlayerProfile> playerProfiles = new ArrayList<PlayerProfile>();

	public PlayerProfile getPlayerProfile(long uuid) {
		
		PlayerProfile result = null;
		for (PlayerProfile pp : playerProfiles) {
			if (pp.uuid == uuid)
			{
				result = pp;
				break;
			}
		}
		
		if (result == null) {
			
			//Look if the player is already logged in
			Player player = plugin.getServer().getPlayerByUUID(uuid);
			
			//If it is load it
			if (player != null) {
				PlayerProfile pp = new PlayerProfile(uuid, player.getName());
				playerProfiles.add(pp);
				return pp;
			}
			//If it isn't, load it but don't assume the name
			else
			{
				PlayerProfile pp = new PlayerProfile(uuid);
				playerProfiles.add(pp);
				return pp;
			}
		}
		
		return result;
	}

	public void removePlayerProfile(long uuid) {
		
		List<PlayerProfile> profilesToDelete = new ArrayList<PlayerProfile>();
		for (PlayerProfile pp : playerProfiles) {

			if (pp.uuid == uuid)
			{
				pp.saveProfile();
				profilesToDelete.add(pp);
			}
		}

		for (PlayerProfile delete : profilesToDelete) {
			playerProfiles.remove(delete);
		}
	}

	public void addPlayerProfile(long uuid, String name) {
		PlayerProfile add = new PlayerProfile(uuid, name);
		playerProfiles.add(add);
	}

	public void saveAll() {
		for (PlayerProfile pp : playerProfiles) {
			pp.saveProfile();
		}
	}
}

File: plugin/src/main/java/io/xol/z/plugin/player/PlayerProfiles.kt
package io.xol.z.plugin.player

import java.util.ArrayList

import io.xol.chunkstories.api.player.Player
import io.xol.z.plugin.XolioZPlugin

class PlayerProfiles(private val plugin: XolioZPlugin) {

    internal var playerProfiles: MutableList<PlayerProfile> = ArrayList()

    fun getPlayerProfile(uuid: Long): PlayerProfile {

        var result: PlayerProfile? = null
        for (pp in playerProfiles) {
            if (pp.uuid == uuid) {
                result = pp
                break
            }
        }

        if (result == null) {

            //Look if the player is already logged in
            val player = plugin.server.getPlayerByUUID(uuid)

            //If it is load it
            if (player != null) {
                val pp = PlayerProfile(uuid, player.name)
                playerProfiles.add(pp)
                return pp
            } else {
                val pp = PlayerProfile(uuid)
                playerProfiles.add(pp)
                return pp
            }//If it isn't, load it but don't assume the name
        }

        return result
    }

    fun removePlayerProfile(uuid: Long) {

        val profilesToDelete = ArrayList<PlayerProfile>()
        for (pp in playerProfiles) {

            if (pp.uuid == uuid) {
                pp.saveProfile()
                profilesToDelete.add(pp)
            }
        }

        for (delete in profilesToDelete) {
            playerProfiles.remove(delete)
        }
    }

    fun addPlayerProfile(uuid: Long, name: String) {
        val add = PlayerProfile(uuid, name)
        playerProfiles.add(add)
    }

    fun saveAll() {
        for (pp in playerProfiles) {
            pp.saveProfile()
        }
    }
}
